Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -22,9 +41,449 @@ Digitizer::~Digitizer()
 
 void Digitizer::process(std::vector<o2::trd::HitType> const& hits, std::vector<o2::trd::Digit>& digits)
 {
-  // very basic implementation
-  // for every hit seen, create a digit
-  for (auto& hit : hits) {
+  // (WIP) Implementation for digitization
+
+  // Check if Geometry and if CCDB are available as they will be requiered
+  /*
+   mGeom = new TRDGeometry();
+   TRDCalibDB * calibration = new TRDCalibDB();
+   const int nTimeBins = calibration->GetNumberOfTimeBinsDCS();
+  */
+
+  // Create a copy of the Hit vector and sort its elements according to the detector number
+  // I was thinking about making a list and push ordered items at hit creation:
+  // Creating an ordered list + casting a list to vector is O(n) and O(n), then in total O(n^2),
+  // Pushing hits to vector + sorting vector is O(1) and O(n*log(n)), so O(n*log(n)) in total
+  // std::vector<o2::trd::HitType> hitCont = hits;
+  // if (!SortHits(hitCont)) {
+  //   LOG(FATAL) << "Hits sorting failed" << FairLogger::endl;
+  //   return;
+  // }
+
+  // Loop over all TRD detectors
+  // Get the a hit container for all the hits in a given detector then call ConvertHits for a given detector (0 - 539)
+  const int kNdet = 540; // Get this from TRD Geometry
+  int totalNumberOfProcessedHits = 0;
+  LOG(INFO) << "Start of processing " << hits.size() << " hits" << FairLogger::endl;
+  for (int det = 0; det < kNdet; ++det) {
+    // Loop over all TRD detectors
+
+    // Jump to the next detector if the detector is
+    // switched off, not installed, etc
+    /*
+    if (calibration->IsChamberNoData(det)) {
+      continue;
+    }
+    if (!mGeo->ChamberInGeometry(det)) {
+      continue
+    }
+    */
+
+    // Hits are sorted for each detector, but you dont know how many hits each detector got
+    // if (!GetHitContainer(det, hitCont, hitContInDet) ||
+    //     !CheckHitContainer(det, hitContInDet)) {
+    //   break;
+    // }
+
+    // Skip detectors without hits
+    std::vector<o2::trd::HitType> hitContainer;
```

Comment:
probably better to use a (cached) member container instead of creating new container within loop each time

Reply:
I thought about the same and I tried, but ended up with a few problems and bugs. I'll leave this implementation for later. Now, I just moved `hitContainer` as a member of the class  `mHitContainer`.